client/client.lua

```diff
@@ -43,12 +44,14 @@ function updateData(name, value)
 end
 
 function doVehicleUpdate()
-    local vehicle = GetVehiclePedIsIn(GetPlayerPed(-1), 0)
+    local vehicle = GetVehiclePedIsIn(PlayerPedId(), 0)
 
     if temp["vehicle"] ~= vehicle and vehicle ~= 0 then
         -- Update it
         local vehicleClass = GetVehicleClass(vehicle)
-        updateData("Vehicle", GetDisplayNameFromVehicleModel(GetEntityModel(vehicle)))
+        
+        -- Added GetLabelText() to the vehicle display name to convert the vehicle name to a nicer version.
```

Comment:
The only problem I can see with this (well, more of a nit pick really) is that this would be localised to the client running the game then, this localised string is sent to the map. So, this means a Chinese player will send a Chinese string to the interface.

Reply:
Oh really? I thought the function always returned these global names (for any language):
https://pastebin.com/wvpyS4kS
and **only** if the language is set to **English** will it return these:
https://pastebin.com/dA3TbkZw
Didn't know they fully supported localization for other languages...

Anyway in that case just remove that change.

Reply:
> Oh really? I thought the function always returned these global names (for any language):
> https://pastebin.com/wvpyS4kS

Yea it does. You're then putting this through the `GetLabelText` native which, translates the global name to a localised name (I've [got a tutorial on labels which kind of explains it](https://forum.fivem.net/t/how-to-using-labels/53198), shameless plug is shameless). In this case the "global" name is the label.

> and only if the language is set to English will it return these:
> https://pastebin.com/dA3TbkZw

No, there's 11 other languages that are supported by GTA :D

Reply:
Good to know :)